src/execution/compiler/operator/sort_translator.cpp

```diff
@@ -200,14 +225,36 @@ void SortTopTranslator::DeclareIterator(FunctionBuilder *builder) {
 }
 
 void SortTopTranslator::GenForLoop(FunctionBuilder *builder) {
-  // for (; @sorterIterHasNext(&sort_iter); @sorterIterNext(&sort_iter))
+  if (use_limit_) {
```

Comment:
You don't need a limit variable. The `Sorter` will handle this for you if you've performed a top-k sort. You just need to check `op->GetOffset()` to ensure sending tuples after `offset` tuples:

```
for (; @sorterIterHasNext(); @sorterIterNext()) {
  if (offset < num_tuples) {
    // Consume
  }
  num_tuples = num_tuples + 1
}
```